File: 品味Java的21種設計模式/dp_java/bridge/example6/CommonMessage.java
package cn.javass.dp.bridge.example6;
/**
 * 普通消息
 */
public class CommonMessage extends AbstractMessage{
	public CommonMessage(MessageImplementor impl) {
		super(impl);
	}

	public void sendMessage(String message, String toUser) {
		//对于普通消息，什么都不干，直接调用父类的方法，把消息发送出去就可以了
		super.sendMessage(message, toUser);
	}	
}

File: 品味Java的21種設計模式/dp_cpp/bridge/example6/CommonMessage.cpp
#include "CommonMessage.h"

namespace cn {
namespace javass {
namespace dp {
namespace bridge {
namespace example6 {

CommonMessage::CommonMessage(MessageImplementor *impl) : AbstractMessage(impl) {
}

void CommonMessage::sendMessage(std::string message, std::string toUser) {
    //对于普通消息，什么都不干，直接调用父类的方法，把消息发送出去就可以了
    AbstractMessage::sendMessage(message, toUser);
}
}
}
}
}
}
